src/main/java/spoon/support/JavaOutputProcessor.java

```diff
@@ -44,19 +46,28 @@
  */
 public class JavaOutputProcessor extends AbstractProcessor<CtNamedElement> implements FileGenerator<CtNamedElement> {
 	PrettyPrinter printer;
-
-	File directory;
+	private Environment environment;
 
 	List<File> printedFiles = new ArrayList<>();
 
+	public JavaOutputProcessor(PrettyPrinter printer) {
+		this.printer = printer;
```

Comment:
that does not seem nice :/

Reply:
You're talking about the line below?

Reply:
No the constructor

Reply:
Could you elaborate ?

Reply:
why did you need to copy the env?
it is accessible via getFactory().getEnv().